### DISCUSSION TITLE
x-www-form-urlencoded requests not working on Android

### DISCUSSION THREAD

**Discussion by M-Lanza** [ionic-team/capacitor#6107] [Q&A] [answered]
I recently enabled the Capacitor Http plugin as stated in the documentation and i'm running into some problems on Android.
When I post a request with `x-www-form-urlconded` as Content-Type using the Angular http client, the Capacitor Plugin seems to not be able to cast the HttpParams to a JSObject.

Similar issue (if not the same) was also reported on the capacitor-community/http repo: [here](https://github.com/capacitor-community/http/issues/270)
There was suggested to use and object, but unfortunately i'm running on the same problem.

Any ideas or suggestions? 🙏 

Dependencies:
    "@angular/common": "^14.0.0",
    "@angular/core": "^14.0.0",
    "@angular/forms": "^14.0.0",
    "@angular/platform-browser": "^14.0.0",
    "@angular/platform-browser-dynamic": "^14.0.0",
    "@angular/router": "^14.0.0",
    "@capacitor/android": "4.4.0",
    "@capacitor/app": "4.1.0",
    "@capacitor/core": "4.5.0",
    "@capacitor/haptics": "4.0.1",
    "@capacitor/keyboard": "4.0.1",
    "@capacitor/status-bar": "4.0.1",
    "@ionic/angular": "^6.1.9"

Capacitor Http Plugin StackTrace:
`E/Capacitor/Plugin: JSONException
    org.json.JSONException: JSValue could not be coerced to JSObject.
        at com.getcapacitor.JSValue.toJSObject(JSValue.java:39)
        at com.getcapacitor.plugin.util.CapacitorHttpUrlConnection.setRequestBody(CapacitorHttpUrlConnection.java:194)
        at com.getcapacitor.plugin.util.HttpRequestHandler.request(HttpRequestHandler.java:399)
        at com.getcapacitor.plugin.CapacitorHttp$1.run(CapacitorHttp.java:34)`

**Comment by jcesarmobile** [comment, marked as answer]
The issue has been reported here https://github.com/ionic-team/capacitor/issues/5996
